### PR TITLE
Autonomous Integration

### PR DESCRIPTION
**Pull request by StephenWelch**

### CODE REVIEW

File: common/src/main/java/us/ilite/common/lib/util/Conversions.java [hunk 1]
```diff
@@ -32,6 +32,10 @@ public static double ticksToInches(int ticks) {
         return ticksToRotations(ticks) * SystemSettings.kDriveWheelCircumference;
     }
 
+    public static int inchesToTicks(double inches) {
```
Comment (kc5tjn) [inline common/src/main/java/us/ilite/common/lib/util/Conversions.java:35]: Seems like a very generic name for a very specific conversion, or is this conversion applicable for more than the drive train?

File: robot/src/main/java/us/ilite/robot/Robot.java [hunk 2]
```diff
@@ -49,22 +41,22 @@
...
-    private Superstructure mSuperstructure = new Superstructure();
+    private CommandManager mAutonomousCommandManager = new CommandManager();
+    private CommandManager mTeleopCommandManager = new CommandManager();
     private DriveController mDriveController = new DriveController(new StrongholdProfile());
     private Drive mDrive = new Drive(mData, mDriveController);
     private Elevator mElevator = new Elevator(mData);
     private HatchFlower mHatchFlower = new HatchFlower();
 
-    private DriverInput mDriverInput = new DriverInput(mDrive, mElevator, mHatchFlower, mSuperstructure, mData);
+    private DriverInput mDriverInput = new DriverInput(mDrive, mElevator, mHatchFlower, mAutonomousCommandManager, mTeleopCommandManager, mData);
```
Comment (kc5tjn) [inline robot/src/main/java/us/ilite/robot/Robot.java]: Be careful with this merge, I've seen inputs to the constructor dropped in earlier merges.

File: robot/src/main/java/us/ilite/robot/Robot.java [hunk 3]
```diff
@@ -114,16 +102,16 @@ public void autonomousInit() {
 
         mSettings.loadFromNetworkTables();
 
+        mLoopManager.setRunningLoops(mDrive);
+        mLoopManager.start();
+
         // Init modules after commands are set
-        mRunningModules.setModules(mSuperstructure);
+        mRunningModules.setModules(mAutonomousCommandManager, mTeleopCommandManager);
```
Comment (kc5tjn) [inline robot/src/main/java/us/ilite/robot/Robot.java]: Was mSuperstructure removed for testing?

File: robot/src/main/java/us/ilite/robot/auto/AutonomousRoutines.java [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+
+    public AutonomousRoutines(TrajectoryGenerator pTrajectoryGenerator, Data pData, Drive pDrive) {
+        mTrajectoryGenerator = pTrajectoryGenerator;
+        mData = pData;
+        mDrive = pDrive;
+
+        mMiddleToMiddleCargoToSideRocket = new MiddleToMiddleCargoToSideRocket(mTrajectoryGenerator);
+    }
+
+    public void generateTrajectories() {
```
Comment (kc5tjn) [inline robot/src/main/java/us/ilite/robot/auto/AutonomousRoutines.java:48]: Seems to be a very generic name for very specific trajectories

**Review by DanielTolbert — DISMISSED** [review]

**Review by DanielTolbert — DISMISSED** [review]

**Pull request merged**